lib/lib.go

```diff
@@ -833,11 +833,40 @@ func (inst *Instance) Peer() PeerMethods {
 	return PeerMethods{d: inst}
 }
 
+// SQL returns the SQLMethods that Instance has registered
+func (inst *Instance) SQL() SQLMethods {
+	return SQLMethods{d: inst}
+}
+
 // Transform returns the TransformMethods that Instance has registered
 func (inst *Instance) Transform() TransformMethods {
 	return TransformMethods{d: inst}
 }
 
+// WithSource returns a wrapped instance that will resolve refs from the given source
+func (inst *Instance) WithSource(source string) *InstanceSourceWrap {
+	return &InstanceSourceWrap{
+		source: source,
+		inst:   inst,
+	}
+}
+
+// InstanceSourceWrap is a wrapped instance with an explicit resolver source added
+type InstanceSourceWrap struct {
+	source string
+	inst   *Instance
+}
```

Comment:
Yeah in my head `WithConfig` would construct shallow-copy `*Instance` . Specifically a method like `WithConfig` should  _not_ construct new copies of subsystems. With access to subsystems governed by scope I think that's a safe approach to take.

As for the config change, I vey much see your point on adding a `Source` field to config, and think your mental model of it being an "Option" according to the parlance laid out here: https://github.com/qri-io/qri/blob/9f36fc56b1bba27d3875f4418ec73020c4efad40/lib/lib.go#L63-L70

To capture setting both "Config" and "Options", we could flip to a `WithOptions` method instead of a `WithConfig`, and use the `OptConfig` function to override configurations.

But now we're actively modeling a different issue, so I'm good to merge this as-is & think we should move this discussion to an issue.